models/gpt2/example_generate.py

```diff
@@ -0,0 +1,59 @@
+import jax
+import jax.numpy as jnp
+
+import os
+import sys
+
+from transformers import GPT2Tokenizer
+
+tknzr = GPT2Tokenizer.from_pretrained("gpt2")
+
+import model
+import config
+
+cfg = config.get_config()
+B = cfg.B  # Batch size
+K = cfg.K  # Input sequence length
+S = cfg.S
+T = cfg.T  # Batched decode
+L, _, _, Q, H, _ = model.model_sizes["gpt2"]
+
+
+def generate_text(params, ids, n_new_words):
+    nids = len(ids)
+    t = jnp.asarray([nids], dtype=jnp.int32)
+    prompt = jnp.asarray([ids + [0] * (K - nids)], dtype=jnp.int32)
+
+    kv = model.init_kv(1, S, L, Q, H, dtype=jnp.float32)
+    kv = [jnp.tile(k, (1, prompt.shape[0], 1, 1, 1)) for k in kv]
+    kv, x = model.encode(params, kv, prompt, 0, t)
+
+    ret = []
+    ret.append(x.item())
+
+    for _ in range(min(n_new_words, S - nids)):
+        kv, x = model.decode(params, kv, x, t)
+        t = t + 1
+        if x.item() == 50256:
+            break
```

Comment:
Don't quite know where this constant is coming from. Could you add comment?

Reply:
Thaks for the reminder! Done!